Allow package installation of config package (with yaml files) without the need to have AC Tool package pre-installed

Big fan of your project - love it!

A pain point I'm running into is the installation of ACTool using the content-package-maven-plugin (I want to avoid installing the package via the UI). 

I've specified the following dependency:
```
<dependencies>
    <dependency>
        <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
	<artifactId>accesscontroltool-package</artifactId>
	<version>2.0.1</version>
	<type>zip</type>
	<exclusions>
	    <exclusion>
	        <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
	        <artifactId>accesscontroltool-bundle</artifactId>
	    </exclusion>
        </exclusions>
    </dependency>
</dependencies>
```
And the following plugin configuration:
```
<plugin>
    <groupId>com.day.jcr.vault</groupId>
    <artifactId>content-package-maven-plugin</artifactId>
    <extensions>true</extensions>
    <configuration>
        <group>mygroup</group>
        <useProxy>false</useProxy>
        <filterSource>src/main/content/META-INF/vault/filter.xml</filterSource>
        <targetURL>http://${sling.host}:${sling.port}/crx/packmgr/service.jsp</targetURL>
       <properties>
            <installhook.actool.class>biz.netcentric.cq.tools.actool.installhook.AcToolInstallHook</installhook.actool.class>
       </properties>
       <subPackages>
           <subPackage>
                <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
                <artifactId>accesscontroltool-package</artifactId>
                <filter>true</filter>
           </subPackage>
      </subPackages>
    </configuration>
</plugin>
```
However, the hook executes before the sub package is installed so the groups/ACLs are not applied. 
```
*INFO* [qtp282918978-3407] org.apache.jackrabbit.vault.packaging.impl.InstallHookProcessorImpl Loading Hook actool: Main-Class = biz.netcentric.cq.tools.actool.installhook.AcToolInstallHook
*ERROR* [qtp282918978-3407] org.apache.jackrabbit.vault.packaging.impl.InstallHookProcessorImpl Error while initializing hook: org.apache.jackrabbit.vault.packaging.PackageException: hook's main class biz.netcentric.cq.tools.actool.installhook.AcToolInstallHook not found: actool
```
Have you attempted to install ACTool this way? What would you recommend?

Another reason to have this is that for the upgrade case, it is possible to get the following error (if everything is included in one big complete package, I had had this even though the contained packages had correct dependencies defined, but those are only package dependencies and don't help to ensure the sync bundle installation is done)

```
[ERROR] Request failed: org.apache.jackrabbit.vault.packaging.PackageException: hook's main class biz.netcentric.cq.tools.actool.installhook.AcToolInstallHook not found: actool (500)
[WARNING] Deployment of artifact projectGroupId:project-configuration-package:2.16.0-SNAPSHOT:zip failed with exeception: Error while installing package. Check log for details.
```

--- a/accesscontroltool-package/pom.xml
+++ b/accesscontroltool-package/pom.xml
@@ -70,10 +70,11 @@
                         </goals>
                         <configuration>
                             <artifactItems>
+                                <!-- See http://sling.apache.org/documentation/bundles/installer-provider-installhook.html -->
                                 <artifactItem>
                                     <groupId>org.apache.sling</groupId>
                                     <artifactId>org.apache.sling.installer.provider.installhook</artifactId>
-                                    <version>1.0.0-SNAPSHOT</version>
+                                    <version>1.0.0</version>
                                 </artifactItem>
                             </artifactItems>
                             <outputDirectory>${project.build.directory}/vault-work/META-INF/vault/hooks</outputDirectory>
@@ -91,7 +92,8 @@
                     <filterSource>src/main/META-INF/vault/filter.xml</filterSource>
                     <properties>
                         <acHandling>merge_preserve</acHandling>
-                        <installPathRegex>/apps/netcentric/actool/install.*</installPathRegex>
+                        <installPathRegex>/apps/netcentric/actool/.*</installPathRegex>
+                        <waitForOsgiEventsQuietInSec>5</waitForOsgiEventsQuietInSec>
                     </properties>
                     <embeddeds>
                         <embedded>

--- a/accesscontroltool-package/pom.xml
+++ b/accesscontroltool-package/pom.xml
@@ -69,7 +69,9 @@
                     <filterSource>src/main/META-INF/vault/filter.xml</filterSource>
                     <properties>
                         <acHandling>merge_preserve</acHandling>
-                        <installPathRegex>/apps/netcentric/actool/.*</installPathRegex>
+                        <!-- install the two bundles and service user config synchronously in order to be callable from within 
+                              same complete package (if yaml file containing package declares the dependency to it) -->
+                        <installPathRegex>/apps/netcentric/actool/(config|install)/.*</installPathRegex>
                         <waitForOsgiEventsQuietInSec>5</waitForOsgiEventsQuietInSec>
                     </properties>
                     <embeddeds>

--- a/accesscontroltool-package/pom.xml
+++ b/accesscontroltool-package/pom.xml
@@ -59,6 +59,30 @@
             </plugin>
 
             <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>copy-hook-into-package</id>
+                        <phase>generate-resources</phase>
+                        <goals>
+                            <goal>copy</goal>
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>org.apache.sling</groupId>
+                                    <artifactId>org.apache.sling.installer.provider.installhook</artifactId>
+                                    <version>1.0.0-SNAPSHOT</version>
+                                </artifactItem>
+                            </artifactItems>
+                            <outputDirectory>${project.build.directory}/vault-work/META-INF/vault/hooks</outputDirectory>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <plugin>
                 <groupId>com.day.jcr.vault</groupId>
                 <artifactId>content-package-maven-plugin</artifactId>
                 <extensions>true</extensions>
@@ -67,6 +91,7 @@
                     <filterSource>src/main/META-INF/vault/filter.xml</filterSource>
                     <properties>
                         <acHandling>merge_preserve</acHandling>
+                        <installPathRegex>/apps/netcentric/actool/install.*</installPathRegex>
                     </properties>
                     <embeddeds>
                         <embedded>

